lib/analyzer/applications/java.ts

```diff
@@ -3,40 +3,34 @@ import * as path from "path";
 import { bufferToSha1 } from "../../buffer-utils";
 import { JarFingerprintsFact } from "../../facts";
 import { JarFingerprint } from "../types";
-import { JarBuffer, JarCoords, JarInfo } from "./types";
+import { AggregatedJars, JarBuffer, JarCoords, JarInfo } from "./types";
 import { AppDepsScanResultWithoutTarget, FilePathToBuffer } from "./types";
 
+/**
+ * @param {{[fileName: string]: Buffer}} input fileName
+ * @returns {AggregatedJars}
+ */
 function groupJarFingerprintsByPath(input: {
   [fileName: string]: Buffer;
-}): {
-  [path: string]: JarBuffer[];
-} {
-  const jarFingerprints: JarBuffer[] = Object.entries(input).map(
-    ([filePath, buffer]) => {
-      return {
-        location: filePath,
-        buffer,
-        coords: null,
-        dependencies: [],
-        nestedJars: [],
-      };
-    },
-  );
-
-  const resultAggregatedByPath = jarFingerprints.reduce(
-    (jarsAggregatedByPath, jarFingerprint) => {
-      const location = path.dirname(jarFingerprint.location);
-      jarsAggregatedByPath[location] = jarsAggregatedByPath[location] || [];
-      jarsAggregatedByPath[location].push(jarFingerprint);
-      return jarsAggregatedByPath;
-    },
-    {},
-  );
+}): AggregatedJars {
+  const resultAggregatedByPath: AggregatedJars = {};
+  Object.keys(input).forEach((filePath) => {
```

Comment:
Good job on removing `reduce`! Is there a reason you chose to replace `map` with `forEach`? Asking because it seemed more convenient to use map when when I looked at `buffer`.

Reply:
It seems that in this case, it acts the same, map allocates a new struct and in this case, I thought it is unnecessary (would love to hear if I'm missing anything)

Reply:
It just seemed a little more readable to me. Especially since you are looking at the value of `buffer`.